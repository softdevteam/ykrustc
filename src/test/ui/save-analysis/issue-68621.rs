// compile-flags: -Zsave-analysis

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete

trait Trait {}

trait Service {
    type Future: Trait;
}

struct Struct;

impl Service for Struct {
    type Future = impl Trait; //~ ERROR: could not find defining uses
}

fn main() {}
