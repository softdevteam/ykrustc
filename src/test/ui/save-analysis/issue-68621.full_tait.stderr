warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/issue-68621.rs:5:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error: could not find defining uses
  --> $DIR/issue-68621.rs:17:19
   |
LL |     type Future = impl Trait;
   |                   ^^^^^^^^^^

error: aborting due to previous error; 1 warning emitted

